mlflow/sklearn/utils.py

```diff
@@ -524,19 +523,24 @@ def _log_specialized_estimator_content(
             _logger.warning(msg)
             return
 
-        _matplotlib_config = {"savefig.dpi": 175, "figure.autolayout": True, "font.size": 8}
-
-        with TempDir() as tmp_dir, matplotlib.rc_context(_matplotlib_config):
+        with TempDir() as tmp_dir:
             for artifact in artifacts:
                 try:
-                    display = artifact.function(**artifact.arguments)
-                    display.ax_.set_title(artifact.title)
-                    artifact_path = "{}.png".format(artifact.name)
-                    filepath = tmp_dir.path(artifact_path)
-                    display.figure_.savefig(fname=filepath, format="png")
+                    import matplotlib
                     import matplotlib.pyplot as plt
 
-                    plt.close(display.figure_)
+                    _matplotlib_config = {
+                        "savefig.dpi": 175,
+                        "figure.autolayout": True,
+                        "font.size": 8,
+                    }
+                    with matplotlib.rc_context(_matplotlib_config):
+                        display = artifact.function(**artifact.arguments)
+                        display.ax_.set_title(artifact.title)
+                        artifact_path = "{}.png".format(artifact.name)
+                        filepath = tmp_dir.path(artifact_path)
+                        display.figure_.savefig(fname=filepath, format="png")
+                        plt.close(display.figure_)
```

Comment:
```suggestion
                        plt.close(display.figure_)
                    except ImportError:
                        # warning message
                        break
```

@fa9r Thanks for the PR! Can we break if we fail to import `matplotlib`?

Reply:
Added break for ImportErrors in Exception block below

Reply:
Can we separate except branches?

```python
try:
   ...
except ImportError as ie:
   ...
except Exception as e:
    ...
```

Reply:
Sure. That is indeed more readable, good catch.